src/components/modules/caret.ts

```diff
@@ -428,11 +233,23 @@ export default class Caret extends Module {
   public navigateNext(force = false): boolean {
     const { BlockManager } = this.Editor;
     const { currentBlock, nextBlock } = BlockManager;
-    const { nextInput } = currentBlock;
-    const isAtEnd = this.isAtEnd;
+
+    if (currentBlock === undefined) {
+      return false;
+    }
+
+    const { nextInput, currentInput } = currentBlock;
+    const isAtEnd = currentInput !== undefined ? caretUtils.isAtEndOfInput(currentInput) : undefined;
```

Comment:
Similar expression is used several times, maybe worth to extract it to a function. Or allow caret utils to accept undefined

Reply:
I believe that utility methods like `isAtStartOfInput()` should not handle case when input is `undefined `. Module should explicitly handle the cases when block has no inputs or there is no `currentBlock` at all since it is a part of business logic.